src/mqt/ddsim/sampler.py

```diff
@@ -0,0 +1,144 @@
+
+class DDSIMBackendSampler(BaseSampler):
+    _BACKEND = QasmSimulatorBackend()
+
+    def __init__(
+        self,
+        options: dict | None = None,
+        bound_pass_manager: PassManager | None = None,
+        skip_transpilation: bool = False,
+    ):
+        """Initialize a new DDSIM Sampler
+
+        Args:
+            options: Default options.
+            bound_pass_manager: An optional pass manager to run after
+                parameter binding.
+            skip_transpilation: If this is set to True the internal compilation
+                of the input circuits is skipped and the circuit objects
+                will be directly executed when this objected is called.
+        Raises:
+            ValueError: If backend is not provided
+        """
+
+        super().__init__(options=options)
+        self._transpile_options = Options()
+        self._bound_pass_manager = bound_pass_manager
+        self._preprocessed_circuits: list[QuantumCircuit] | None = None
+        self._transpiled_circuits: list[QuantumCircuit] = []
+        self._skip_transpilation = skip_transpilation
+
+    @property
+    def transpiled_circuits(self) -> list[QuantumCircuit]:
+        if self._skip_transpilation:
+            self._transpiled_circuits = list(self._circuits)
+        elif len(self._transpiled_circuits) < len(self._circuits):
+            # transpile only circuits that are not transpiled yet
+            self._transpile()
+        return self._transpiled_circuits
+
+    @property
+    def backend(self):
+        return self._BACKEND
+
+    @property
+    def transpile_options(self) -> Options:
+        return self._transpile_options
+
+    def set_transpile_options(self, **fields):
+        self._transpile_options.update_options(**fields)
+
+    def _transpile(self):
+        from qiskit import transpile
+
+        start = len(self._transpiled_circuits)
+        self._transpiled_circuits.extend(
+            transpile(
+                self._circuits[start:],
+                target=self.backend.target,
+                **self.transpile_options.__dict__,
+            ),
+        )
+
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Parameters],
+        **run_options,
+    ) -> PrimitiveJob:
+        for circuit in circuits:
+            self._circuits.append(circuit)
+            self._parameters.append(circuit.parameters)
+
+        job = PrimitiveJob(self._call, parameter_values, **run_options)
+        job.submit()
+        return job
+
+    def _call(
+        self,
+        parameter_values: Sequence[Parameters],
+        **run_options,
+    ) -> SamplerResult:
+        transpiled_circuits = self.transpiled_circuits
+        bound_circuits = self.backend.assign_parameters_from_backend(transpiled_circuits, parameter_values)
+        bound_circuits = self._bound_pass_manager_run(bound_circuits)
+
+        result = [self.backend.run(bound_circuits, **run_options).result()]
```

Comment:
Yes. I just put it explicitly to adapt the code to the bound_pass_manager method. 
But since we are removing that, I can move the parameter binding task to the backend.